packages/react-core/src/components/TreeView/TreeViewListItem.tsx

```diff
@@ -186,20 +198,22 @@ const TreeViewListItemBase: React.FunctionComponent<TreeViewListItemProps> = ({
               onClick={(evt: React.MouseEvent) => {
                 if (!hasCheck) {
                   onSelect && onSelect(evt, itemData, parentItem);
-                  if (children && evt.isDefaultPrevented() !== true) {
+                  if (!hasSelectableNodes && children && evt.isDefaultPrevented() !== true) {
                     setIsExpanded(!internalIsExpanded);
                   }
                 }
               }}
-              tabIndex={-1}
+              tabIndex={hasSelectableNodes ? 1 : -1}
```

Comment:
```suggestion
              tabIndex={hasSelectableNodes ? 0 : -1}
```

Though it looks like this tab index is being overridden anyway by TreeViewRoot (line 38).